tests/test_vega.py

```diff
@@ -263,7 +256,776 @@ def test_fill_anchor_in_string(tmp_dir):
+
+    assert plot_content["data"]["values"] == expected_datapoints
+    assert plot_content["facet"]["row"] == row_encoding
+    assert plot_content["spec"]["transform"][0]["groupby"] == [y, "actual"]
+    assert plot_content["spec"]["transform"][1]["groupby"] == group_by_y
+    assert plot_content["spec"]["transform"][2]["groupby"] == group_by_x
+    assert plot_content["spec"]["layer"][0]["width"] == 300
+    assert plot_content["spec"]["layer"][0]["height"] == 300
+
+
+@pytest.mark.parametrize(
+    ",".join(
+        [
+            "datapoints",
+            "y",
+            "anchors_y_definitions",
+            "expected_dp_keys",
+            "shape_encoding",
+            "tooltip_encoding",
+        ]
+    ),
+    (
+        (
+            [
+                {
+                    "rev": "B",
+                    "acc": "0.05",
+                    "other": "field",
+                    "filename": "test",
+                    "field": "acc",
+                    "loss": 0.1,
+                },
+                {
+                    "rev": "C",
+                    "acc": "0.1",
+                    "other": "field",
+                    "filename": "test",
+                    "field": "acc",
+                    "loss": 2,
+                },
+            ],
+            "acc",
+            [{"filename": "test", "field": "acc"}],
+            ["rev", "acc", "other", "loss"],
+            {},
+            [{"field": "rev"}, {"field": "loss"}, {"field": "acc"}],
+        ),
+        (
+            [
+                {
+                    "rev": "B",
+                    "dvc_inferred_y_value": "0.05",
+                    "test_acc": "0.05",
+                    "train_acc": "0.06",
+                    "other": "field",
+                    "filename": "data",
+                    "field": "test_acc",
+                    "loss": 0.1,
+                },
+                {
+                    "rev": "B",
+                    "dvc_inferred_y_value": "0.06",
+                    "test_acc": "0.05",
+                    "train_acc": "0.06",
+                    "other": "field",
+                    "filename": "data",
+                    "field": "train_acc",
+                    "loss": 0.1,
+                },
+                {
+                    "rev": "C",
+                    "dvc_inferred_y_value": "0.1",
+                    "train_acc": "0.1",
+                    "test_acc": "0.2",
+                    "other": "field",
+                    "filename": "train_acc",
+                    "field": "acc",
+                    "loss": 2,
+                },
+                {
+                    "rev": "C",
+                    "dvc_inferred_y_value": "0.2",
+                    "train_acc": "0.1",
+                    "test_acc": "0.2",
+                    "other": "field",
+                    "filename": "test_acc",
+                    "field": "acc",
+                    "loss": 2,
+                },
```

Comment:
I'm confused by this one. Why does rev B have filename `data` but rev C has filenames `train_acc` and `test_acc`?